ludwig/models/modules/convolutional_modules.py

```diff
@@ -530,74 +538,51 @@ def __init__(
-                                             padding='SAME',
-                                             activation=layer['activation'],
-                                             norm=layer['norm'],
-                                             dropout=layer['dropout'],
-                                             dropout_rate=dropout_rate,
-                                             regularizer=regularizer if layer[
-                                                 'regularize'] else None,
-                                             initializer=layer['initializer'],
-                                             is_training=is_training)
-                prev_num_filters = layer['num_filters']
-                logger.debug('  conv_{}: {}'.format(i, layer_output))
+        self.stack = []
 
-                # Pooling
+        for i, layer in enumerate(self.layers):
+            with tf.variable_scope('conv_' + str(i)):
+                # TODO add filter size tuples
+                # TODO provide regularization penalty
+                # TODO provide regularizers
+                # TODO add initializers
+                self.stack.append(
+                    Conv2D(
+                        layer['num_filters'],
+                        layer['filter_size'],
+                        # kernel_regularizer=regularizer if layer['regularize'] else None,
+                    )
+                )
+
+                self.stack.append(Activation(layer['activation']))
+
+                if layer['dropout_rate'] > 0:
+                    self.stack.append(Dropout(layer['dropout_rate']))
+                
                 if layer['pool_size'] is not None:
-                    pool_size = layer['pool_size']
-                    pool_kernel_size = [1, pool_size, pool_size, 1]
-                    pool_stride = layer['pool_stride']
-                    if pool_stride is not None:
-                        pool_kernel_stride = [1, pool_size, pool_size, 1]
-                    else:
-                        pool_kernel_stride = [1, pool_stride, pool_stride, 1]
-                    layer_output = tf.nn.max_pool(
-                        layer_output,
-                        ksize=pool_kernel_size,
-                        strides=pool_kernel_stride,
-                        padding='SAME',
-                        name='pool_{}'.format(i)
+                    self.stack.append(
+                        MaxPool2D(
+                            pool_size=layer['pool_size'],
+                            strides=layer['pool_strides']
+                        )
                     )
 
-                    logger.debug('  pool_{}: {}'.format(
-                        i, layer_output))
+    def call(self, inputs, training=None, mask=None):
+        hidden = inputs
 
-            hidden = layer_output
-            logger.debug('  hidden: {0}'.format(hidden))
+        for layer in self.stack:
+            hidden = layer(hidden, training=training)
```

Comment:
Should probably add mask to this call

Reply:
I'm getting this "TypeError: call() got an unexpected keyword argument 'mask'".
The docs say it should be there but maybe we need to update the version or smth.

Reply:
mmm it could be the case that some layers use it and some don't... which is weird, what's the purpuse to have an interface if you don't respect it? :D Anyway, don't pass it for now, but maybe add a todo so that we will chack this later